extensions/typescript-language-features/src/tsServer/versionManager.ts

```diff
@@ -32,9 +32,16 @@ export class TypeScriptVersionManager extends Disposable {
 		this._currentVersion = this.versionProvider.defaultVersion;
 
 		if (this.useWorkspaceTsdkSetting) {
-			const localVersion = this.versionProvider.localVersion;
-			if (localVersion) {
-				this._currentVersion = localVersion;
+			if (vscode.workspace.trustState === vscode.WorkspaceTrustState.Unknown) {
+				setImmediate(() => {
+					vscode.workspace.requireWorkspaceTrust(false);
```

Comment:
Quick note on the `requireWorkspaceTrust` API: we generally try to have `false` be the default for optional parameters. Can the `modal` parameter be renamed so that its default value can be false instead?

Reply:
One reason we might want to keep it the way it is: As an adopter testing this, its more likely that you want to pop the modal when something is required on demand, so using the api without parameters hits the more common case.

Reply:
Yes but if you rename the parameter so that `false = modal` and `true = no modal`, I think the code would be easier to understand

Even better, pass in an options bad so that I know what this parameter means